### PR TITLE
Translate metadata (title, description)

### PR DESCRIPTION
**Pull request by jasonharrison**
Translate metadata (title, description)

### CODE REVIEW

File: src/App/translations/en.ts [hunk 1]
```diff
@@ -10,6 +10,33 @@ const en = {
...
+    },
+    index: {
+      title: 'Play Free Board Games Online',
+      description:
+        'Play board games in your browser for free. Compete against your online friends or play locally. Free and open-source software project.',
+    },
+    gameTemplate: {
+      title: 'Play Free ${GAME} Online',
+    },
+    chess: {
```
Comment (flamecoals) [inline src/App/translations/en.ts]: this will mix all games in a single file, which is bad. Can move these to a translation file  inside the game folder?

File: src/games/chess/index.ts [hunk 2]
```diff
@@ -17,12 +17,8 @@ export const chessGameDef: IGameDef = {
     { mode: GameMode.OnlineFriend },
     { mode: GameMode.LocalFriend },
   ],
-  description: 'Classic game of Chess',
-  descriptionTag: `Play an online Chess game in your browser against a\
- top chess computer. You can set the computer level from 1 to 8,\
- from easy to grandmaster. You can also easily share a link and play\
- chess with a friend online, or you can share your device and play\
- with a friend locally !`,
+  description: 'metadata.chess.description',
```
Comment (flamecoals) [inline src/games/chess/index.ts]: if this will always be metadata.{code}.description, we dont need this attribute anymore

File: src/games/chess/index.ts [hunk 3]
```diff
@@ -17,12 +17,8 @@ export const chessGameDef: IGameDef = {
...
+  descriptionTag: 'metadata.chess.descriptionTag',
```
Comment (flamecoals) [inline src/games/chess/index.ts]: same here.

File: src/index.html [hunk 4]
```diff
@@ -35,17 +36,14 @@
...
-        typeof navigator !== 'undefined' &&
-        navigator.serviceWorker) {
+      if (typeof window !== 'undefined' && typeof navigator !== 'undefined' && navigator.serviceWorker) {
         navigator.serviceWorker.register('/sw.js').then(function(reg) {
           reg.onupdatefound = function() {
             const installingWorker = reg.installing;
             installingWorker.onstatechange = function() {
-              if (installingWorker.state === 'installed' &&
-                  navigator.serviceWorker.controller) {
+              if (installingWorker.state === 'installed' && navigator.serviceWorker.controller) {
```
Comment (flamecoals) [inline src/index.html]: can you please avoid unrelated changes to the PR? It adds noise to the review. Maybe create another PR for this?

File: src/metadata.ts [hunk 5]
```diff
@@ -9,42 +11,73 @@ export interface IPageMetadata {
...
+    lang = 'en';
+  }
+  registerLang(lang, translations[lang]);
+  const allPagesMetadata = getAllPagesMetadata(lang);
+  let metadata = allPagesMetadata.find(meta => meta.url!.test(url));
   if (!metadata) {
-    return DEFAULT_METADATA;
+    metadata = getDefaultMetadata(lang);
   }
+  metadata.description = trans(metadata.description, lang);
```
Comment (flamecoals) [inline src/metadata.ts]: we dont need to read these if they are always based on the game code

File: src/metadata.ts [hunk 6]
```diff
@@ -9,42 +11,73 @@ export interface IPageMetadata {
...
-  const allPagesMetadata = [...PAGES_METADATA, ...getGamesPageMetadata()];
-  const metadata = allPagesMetadata.find(meta => meta.url!.test(url));
+  let lang = getLangFromURL(url, translations);
+  if (!lang) {
+    // fallback to English
+    lang = 'en';
+  }
+  registerLang(lang, translations[lang]);
+  const allPagesMetadata = getAllPagesMetadata(lang);
+  let metadata = allPagesMetadata.find(meta => meta.url!.test(url));
```
Comment (flamecoals) [inline src/metadata.ts]: This seem to be very wasteful. You want only one page and you are generating the IPageMetadata of everyone. if you execute this to all pages, it will be O(n^2) instead of O(n). 

You could instead change getAllPagesMetadata and getGamesPageMetadata to only be for one page at a time.

File: src/metadata.test.ts [hunk 7]
```diff
@@ -13,25 +13,26 @@ describe('Metadata', () => {
...
+    expect(metadata.description).to.equal('metadata.about.description');
 
     // getGamesPageMetadata()
     metadata = getPageMetadata('/g/chess');
-    expect(metadata.title).to.contain(TITLE_PREFIX);
-
-    expect(metadata.title).to.contain('Chess');
-    expect(metadata.description).to.contain('chess');
+    expect(metadata.title).to.equal(TITLE_PREFIX + 'metadata.gameTemplate.title');
+    expect(metadata.description).to.contain('metadata.chess.descriptionTag');
```
Comment (flamecoals) [inline src/metadata.test.ts]: Why these tests are returning the key and not the translation?

Comment (jasonharrison) [reply src/metadata.test.ts]: Changed to test in English
